src/pilot/assign_role.py

```diff
@@ -1160,11 +1168,46 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
 
                     break
 
+            # Note: This code block represents single disk scenario.
+            raid_disk_ids = []
```

Comment:
Would you mind moving the above line to just after the "if virtual_disk_docs:" line?

Reply:
Shifted `raid_disk_ids` list on when `virtual_disk_docs` exists.